blog/webassembly_interaction_patterns.md

```diff
@@ -0,0 +1,102 @@
+the response via **stdout**.
+
+The primary appeal of this pattern is simplicity and availability. Even languages that struggle to produce freestanding WebAssembly modules (e.g. targeting `wasm32-unknown-unknown`) can often 
+produce `wasm32-wasi` modules. Accepting data via **stdin** means that the WASI plumbing is taking care of marshaling the data through linear memory and the developer doesn't have to do that manually. Returning the result is a simple matter of writing to **stdout**.
+
+The net effect on the developer is a huge boon during today's highly volatile Wasm ecosystem. You can write code that runs both as a regular console app and as a WASI module. This makes
+the code easy to reason about, easy to build, easy to test, and (theoretically) automatically portable to any platform that supports WASI. Sending the request to the module is a simple matter of invoking its default exported function (usually `start` or `_start`).
+
+Here's a few lines of Rust code that can work as easily via `cargo run` as they can as a WASI module:
+```rust
+fn main() -> io::Result<()> {
+    let mut buffer = String::new();
+    let stdin = io::stdin(); 
+    stdin.read_line(&mut buffer)?;
+    println!("Hello, {}!", buffer.trim());
+    Ok(())
+}
+```
+This code reads a line of text from **stdin** and writes a greeting back out. This will work whether you're running it as a CLI app or as a WASI module. In fact, you could simply execute
+this WASI module via the `wasmtime` CLI tool. More common than reading a line, we typically see modules that accept everything available on stdin and then process it, which could mean
+deserializing JSON or some other binary format.
+
+In many command pattern implementations, this module could be started up on demand, process a single request, and then shut down. This is a very common pattern for serverless functions.
+
+## The Reactor Pattern
+It's easy to overthink this one. Rather than being named after some specific library or technology, in this case **reactor** is as its name implies: it _reacts_. 
+The reactor pattern is a command pattern that doesn't shut down after processing a single request. Instead, it stays running and waits for more requests. While both the command and reactor patterns can make use of callbacks for host facilities, these types of callbacks are far more prevalent in reactors. Many implementations of the reactor pattern require that some form of `setup` or `init` function be called prior to the reactor being able to process requests.
+
+Another fundamental difference is that while the command pattern commonly maps a single module to a single function, a reactor can respond to multiple different types of stimulus (which can be modeled as commands, events, or anything else). You'll typically see a dispatcher inside a reactor that routes the request to the appropriate handler. This is a common pattern in many languages and frameworks, but it's especially common in JavaScript and Node.js.
```

Comment:
Comma between that and while - difference is that, while the command pattern....